src/lexer/mod.rs

```diff
@@ -456,20 +460,30 @@ impl Lexer {
     Ok(tokens)
   }
 
-  fn parse_command(&mut self) -> Result<Vec<Token>> {
+  fn parse_command(&mut self, backtick: bool) -> Result<Vec<Token>> {
     let mut tokens = Vec::new();
 
     let mut current_part = EcoString::new();
     let mut line = self.line;
     let mut column = self.column;
+    let mut string_char: Option<char> = None;
+    let mut last_was_whitespace: bool = false;
```

Comment:
What do `string_char` and `last_was_whitespace` do? It's not very clear to me

Reply:
I have added comments to explain the two fields. Basically `string_char` is for remembering the character used to open a string in a command, and `last_was_whitespace` is used for whitespace stripping.